From: Celeste <20312-Celeste@users.gitlab.alpinelinux.org>
Date: Tue, 21 May 2024 07:20:00 +0000
Subject: [PATCH] Remove ancient use Perl 5.8.1 pragma

Perl 5.8.1 was released in 2003, so more than 20 years later,
ensuring we are using at least this version is no longer necessary.

Since Perl 5.39.10, this ancient use pragma is also causing
a test failure in src/pl/plperl/expected/plperl.out:

 -- check that eval is allowed and eval'd restricted ops are caught
 DO $$ eval q{chdir '.';}; warn "Caught: $@"; $$ LANGUAGE plperl;
-WARNING:  Caught: 'chdir' trapped by operation mask at line 1.
+ERROR:  'eval hints' trapped by operation mask at line 1.
+CONTEXT:  PL/Perl anonymous code block

PostgreSQL 16 is not affected as this pragma has been removed in
https://github.com/postgres/postgres/commit/4c1532763a00c21cbb737bc3855e9a31374b119d
which also contains changes to configure scripts, documentation,
readme, and tools. Those changes don't need to be backported here.


--- a/src/pl/plperl/plc_perlboot.pl
+++ b/src/pl/plperl/plc_perlboot.pl
@@ -6,7 +6,6 @@
 use strict;
 use warnings;
 
-use 5.008001;
 use vars qw(%_SHARED $_TD);
 
 PostgreSQL::InServer::Util::bootstrap();
